### PR TITLE
Add the ability to set the maximum number of allowed violations before a build fails

### PR DESCRIPTION
**Pull request by clem0110**
Add the option to allow users to set the maximum number of violations that can occur before a build fails.

### CODE REVIEW

File: src/main/java/org/apache/maven/plugins/pmd/AbstractPmdViolationCheckMojo.java [hunk 1]
```diff
@@ -87,6 +87,14 @@
     @Parameter( property = "pmd.excludeFromFailureFile", defaultValue = "" )
     private String excludeFromFailureFile;
 
+    /**
+     * The maximum number of violations allowed before execution fails.
+     *
+     * @since 3.10
```
Comment (adangel) [inline src/main/java/org/apache/maven/plugins/pmd/AbstractPmdViolationCheckMojo.java]: Minor thing: With 3.9.0 we switched to a 3-part version number, so we should use here 3.10.0.

File: src/test/java/org/apache/maven/plugins/pmd/PmdViolationCheckMojoTest.java [hunk 2]
```diff
@@ -78,6 +78,48 @@ public void testNotFailOnViolation()
...
+                System.out.println( "Caught expected message: " + e.getMessage() );// expected
+            }
+            else
+            {
+                throw new AssertionError( "Expected: '" + message
+                    + "' to contain 'You have 5 PMD violations and 3 warnings.'" );
+            }
+        }
+
+        assertTrue( true );
```
Comment (adangel) [inline src/test/java/org/apache/maven/plugins/pmd/PmdViolationCheckMojoTest.java]: Just remove this line, it's not needed :smile:

**Comment by adangel** [comment]
Created JIRA ticket: https://issues.apache.org/jira/browse/MPMD-256

**Comment by clem0110** [comment]
@adangel I updated those minor things.  Let me know if anything else needs to be done.

**Comment by adangel** [comment]
Thanks for the contribution! It's merged now.

**Pull request merged**